### PR TITLE
feat: [BB-333] implement datepickers

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Implemented Date Pickers and converted it as ISOString, as NMP is using ISOString for their date pickers. For the list, I've narrowed it down and only took Month and day on a default en-us locale.

> [!NOTE]
> You'll notice a quite misalign on the the information buttons in Fields Module. I will fix that on the next styling ticket. Or it might get fixed when the other pr is merged.

### Better Berries
image

### NMP
samplingMonthNMP

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-342-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-342-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrientsList.tsx [hunk 1]
```diff
@@ -84,7 +84,7 @@ const CalculationList: FC<CalculationListProps> = ({ field, cropBalances, result
                   <td>
                     <h4 className="col1">Fertilizer</h4>
                   </td>
-                  <td>
+                  <td aria-hidden="true">
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrientsList.tsx:87]: Why are you adding aria hidden here?

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx [hunk 2]
```diff
@@ -87,7 +87,15 @@ const FieldsListComponent: FC<FieldListProps> = ({ farmDetails }) => {
                 marginRight="50px"
               >
                 <h2>Sampling Month</h2>
-                <p>{HasSoilTest ? SoilTest.sampleDate : ComponentText.NA}</p>
+                <p>
+                  {HasSoilTest
+                    ? new Date(SoilTest.sampleDate).toLocaleDateString('en-US', {
+                        timeZone: 'UTC',
+                        month: 'long',
+                        day: 'numeric',
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx:95]: Could you change this to display month and year so it's the same as nmp? The day really doesn't matter much for the soil tests, it's mainly the month and then the year.

File: frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrients.tsx [hunk 3]
```diff
@@ -396,7 +397,7 @@ const CalculateNutrientsComponent: FC<InputModuleProps> = ({
                       label="Date (optional)"
                       name="applDate"
                       id="applDate"
-                      type="text"
+                      type="date"
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrients.tsx:400]: Nice, didn't know this was an option, good to learn :smile:

**Review by GDamaso — COMMENTED** [review]
Good job!
I think we should improve something before merging though.
When you click the date picker, it only works if you click exactly on the calendar icon. Could you fix this? I think it's important.
Also, it would be better to display the month and year instead of month and day so it matches nmp. The day really doesn't matter that much.

Thanks for the work!

File: frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrientsList.tsx (hunk 1 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrientsList.tsx:87]: It was a eslint error that's blocking me for pushing. Husky issue. Idk actually how it got past yours.

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx (hunk 2 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx:95]: Sure that's easy. I thought nmp was month and day as I've saw on soil test on nmp

File: frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrients.tsx (hunk 3 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/CalculateNutrients/CalculateNutrients.tsx:400]: Yeah! It was something I've saw when googling.

**Review by GDamaso — APPROVED** [review]
Good job man!
:rocket:

**Pull request merged**